server/neptune/workflows/internal/deploy/revision/revision.go

```diff
@@ -102,3 +97,31 @@ func (n *Receiver) Receive(c workflow.ReceiveChannel, more bool) {
 		Tags:           request.Tags,
 	})
 }
+
+func (n *Receiver) createCheckRun(ctx workflow.Context, id, revision string, root activity.Root, repo github.Repo) activities.CreateCheckRunResponse {
+	lock := n.queue.GetLockState()
+	var actions []github.CheckRunAction
+	var summary string
+
+	if lock.Status == queue.LockedStatus && (root.Trigger == activity.MergeTrigger) {
```

Comment:
nit: we can probably get away with not using the `(` operator since `==` has higher precedence over `&&`